pci/src/lib.rs

```diff
@@ -42,3 +50,91 @@ impl PciInterruptPin {
         self as u32
     }
 }
+
+#[derive(Debug)]
+pub enum Error {
+    AllocateGsi,
+    EventFd(io::Error),
+    IrqFd(kvm_ioctls::Error),
+    SetGsiRouting(kvm_ioctls::Error),
+}
+
+// Returns a `Vec<T>` with a size in bytes at least as large as `size_in_bytes`.
+fn vec_with_size_in_bytes<T: Default>(size_in_bytes: usize) -> Vec<T> {
+    let rounded_size = (size_in_bytes + size_of::<T>() - 1) / size_of::<T>();
+    let mut v = Vec::with_capacity(rounded_size);
+    for _ in 0..rounded_size {
+        v.push(T::default())
```

Comment:
If you know T supports Clone: 
`v.resize(rounded_size, T::default())`

Otherwise

`v.resize_with(rounded_size, T::default);`

Reply:
Thanks! Done!